test/PromiseTest.res

```diff
@@ -156,7 +156,7 @@ module Catching = {
     ->then(_ => resolve()) // Since our asyncParse will fail anyways, we convert to promise<unit> for our catch later
     ->catch(e => {
       let success = switch e {
-      | Exn.Error(err) => Exn.message(err) == Some("Unexpected token . in JSON at position 1")
+      | Exn.Error(err) => Exn.name(err) == Some("SyntaxError")
```

Comment:
This was the only thing failing, right, not the others? I'd rather we make only this change then and leave the others unchanged.

Reply:
Why? They're no less terrible. They just haven't broken yet.

Reply:
I don't agree they're terrible. I think they make the test easier to understand because they say what we expect to happen rather than just being a general "TypeError". And if they break they're easy to fix.